public/javascripts/main.js

```diff
@@ -42,18 +42,24 @@ David end
 /**
 Michael start
 */
+const inputusername = document.getElementById('your-username');
 const createbutton = document.getElementById('CreateFormButton');
 if (createbutton) {
   createbutton.onclick = function createURL() {
     location.href = '../party-room';
+    sessionStorage.setItem('name', inputusername.value);
```

Comment:
when executing above line, it will start the navigation. Not sure if there will be a race condition, I would suggest moving the sessionStorage.setItem before `location.href = '../party-room'`